spyder_env_manager/spyder/widgets/packages_table.py

```diff
@@ -250,6 +222,52 @@ def previous_row(self):
             row = rows
         self.selectRow(row - 1)
 
+    def contextMenuEvent(self, event):
+        """Qt Override."""
+        self.context_menu.clear_actions()
+        row = self.rowAt(event.pos().y())
+        packages = self.source_model.packages
+        if packages and packages[row]["requested"]:
+            update_action = self.create_action(
+                self,
+                _("Update package"),
+                triggered=lambda triggered: self.sig_action_context_menu.emit(
+                    EnvironmentPackagesActions.UpdatePackage, packages[row]
+                ),
+                overwrite=True,
+            )
+            uninstall_action = self.create_action(
+                self,
+                _("Uninstall package"),
+                triggered=lambda triggered: self.sig_action_context_menu.emit(
+                    EnvironmentPackagesActions.UninstallPackage, packages[row]
+                ),
+                overwrite=True,
+            )
+            change_action = self.create_action(
+                self,
+                _("Change package version with a constraint"),
+                triggered=lambda triggered: self.sig_action_context_menu.emit(
+                    EnvironmentPackagesActions.InstallPackageVersion, packages[row]
+                ),
+                overwrite=True,
+            )
+            menu_actions = [
```

Comment:
```suggestion

            self.context_menu.clear_actions()
            menu_actions = [
```

Given that the context menu is rebuilt for every package row, I think we need to clear it before adding new actions to it, right?

Reply:
That's done at the begining of the method around line 227

Reply:
Ok, I missed that one. Sorry for the noise.